tests/python/test_ad_basics_fwd.py

```diff
@@ -34,7 +36,46 @@ def test_ad_fwd_multiply():
     def ad_fwd_multiply():
         loss[1] += x[3] * x[4]
 
-    with ti.ad.FwdMode(loss=[loss], parameters=x, seed=[0, 0, 0, 1, 1]):
+    with ti.ad.FwdMode(loss=loss, parameters=x, seed=[0, 0, 0, 1, 1]):
         ad_fwd_multiply()
 
-    assert loss.grad[1] == 7
+    assert loss.dual[1] == 7
+
+
+@test_utils.test(arch=[ti.cpu, ti.gpu])
```

Comment:
OOC: after we switch to lazy_dual approach, do we still need to limit it to cpu &gpu backend?

Reply:
Good catch! Have removed the limit. :)